### PR TITLE
Nxos/input validation

### PR DESCRIPTION
**Pull request by mikewiebe**
### What does this PR do?

This PR adds input validation for a couple of the methods in `salt/modules/nxos.py` and updates the install doc to use `systemctl` to start the nxos guestshell minion.

### Previous Behavior
```
(py2-salt) n9kuser@n9kuser:~/Projects/salt$salt n9k nxos.cmd sendline show version
n9k:
    The minion function caused an exception: Traceback (most recent call last):
      File "/Users/n9kuser/Projects/salt/salt/metaproxy/proxy.py", line 409, in thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "/Users/n9kuser/Projects/salt/salt/executors/direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/modules/nxos.py", line 228, in cmd
        return __salt__[local_command](*args, **kwargs)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/modules/nxos.py", line 335, in sendline
        return __proxy__['nxos.sendline'](command, method, **kwargs)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/proxy/nxos.py", line 323, in sendline
        result = _nxapi_request(command, method, **kwargs)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/proxy/nxos.py", line 538, in _nxapi_request
        data = __utils__['nxos.nxapi_request'](commands, method=method, **conn_args)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/utils/nxos.py", line 313, in nxapi_request
        return client.request(method, commands)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/utils/nxos.py", line 199, in request
        return self.parse_response(response, command_list)
      File "/Users/n9kuser/Virtualenvs/py2-salt/var/cache/salt/proxy/extmods/n9k/utils/nxos.py", line 263, in parse_response
        raise NxosError('Unknown Error: {}, Code: {}'.format(msg, code))
    NxosError: Unknown Error: Message not supported, Code: 501

```
### New Behavior

```
(py2-salt) n9kuser@n9kuser:~/Projects/salt$salt n9k nxos.sendline show version
n9k:
    INPUT ERROR: Second argument 'method' must be one of [u'cli_show_ascii', u'cli_show', u'cli_conf']
      Value passed: version
      Hint: White space separated commands should be wrapped by double quotes

```
Remove this section if not relevant

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/modules/nxos.py [hunk 1]
```diff
@@ -356,6 +363,12 @@ def show(commands, raw_text=True, **kwargs):
         salt '*' nxos.show 'show bgp sessions ; show processes' raw_text=False
         salt 'regular-minion' nxos.show 'show interfaces' host=sw01.example.com username=test password=test
     '''
+    if not isinstance(raw_text, bool):
+        msg = "INPUT ERROR: Second argument 'raw_text' must be either True or False"
+        msg = msg + '\n  Value passed: {}'.format(raw_text)
+        msg = msg + '\n  Hint: White space separated show commands should be wrapped by double quotes'
```
Comment (gtmanfred) [inline salt/modules/nxos.py]: This could just be a multiline stream instead of concatenating everything.

**Review by gtmanfred — CHANGES_REQUESTED** [review]

File: salt/modules/nxos.py [hunk 2]
```diff
@@ -332,9 +332,11 @@ def sendline(command, method='cli_show_ascii', **kwargs):
     '''
     smethods = ['cli_show_ascii', 'cli_show', 'cli_conf']
     if method not in smethods:
-        msg = "INPUT ERROR: Second argument 'method' must be one of {}".format(smethods)
-        msg = msg + '\n  Value passed: {}'.format(method)
-        msg = msg + '\n  Hint: White space separated commands should be wrapped by double quotes'
+        msg = """
+        INPUT ERROR: Second argument 'method' must be one of {}
```
Comment (gtmanfred) [inline salt/modules/nxos.py]: please include the numbers or variable names in the {} brackets.

This is a salt style guide thing.

Comment (mikewiebe) [reply salt/modules/nxos.py]: Fixed

**Review by gtmanfred — APPROVED** [review]

**Comment by mikewiebe** [comment]
@gtmanfred Thanks for the review and approval.  Is there anything else needed for the merge?

**Comment by gtmanfred** [comment]
It would require someone from @saltstack/team-core to merge.

**Comment by mikewiebe** [comment]
@dwoz Reaching out to you because it appears that you have been merging many of the PR's lately.  @gtmanfred approved this PR but not sure if it's waiting on anything else before it can be merged?

**Review by dwoz — APPROVED** [review]

**Pull request merged**